editor/import/3d/resource_importer_scene.cpp

```diff
@@ -2417,6 +2440,14 @@ Node *ResourceImporterScene::_generate_meshes(Node *p_node, const Dictionary &p_
 		mesh_node->set_transform(src_mesh_node->get_transform());
 		mesh_node->set_skin(src_mesh_node->get_skin());
 		mesh_node->set_skeleton_path(src_mesh_node->get_skeleton_path());
+
+		// Copy meta from imported mesh.
+		List<StringName> meta_keys;
+		src_mesh_node->get_meta_list(&meta_keys);
+		for (const StringName &key : meta_keys) {
+			mesh_node->set_meta(key, src_mesh_node->get_meta(key));
+		}
```

Comment:
Looks the same as `_copy_meta` from the GLTF code.

```
static void _copy_meta(Object *p_src, Object *p_dst) {
	List<StringName> meta_keys;
	p_src->get_meta_list(&meta_keys);
	for (const StringName &key : meta_keys) {
		p_dst->set_meta(key, p_src->get_meta(key));
	}
}
```

@akien-mga Thoughts on having this in core? Copying meta seems like a pretty general operation. If not, we can leave the code in this PR as-is, duplicate code is better than an unwanted spaghetti connection.

Reply:
The godot code style allows you to move something to a common place once it is needed in two places.

Try changing the PR to add it as a function on Object. Make sure it is not exposed to users (no need to bind, just declare/implement in c++). There shouldn't need to be discussion as long as it is self contained and internal to the engine

Reply:
I've added it on `Object` but it has to be called something different since the code here knows the target will have no meta, whereas on `Object` it's not so clear cut, so have called it `merge_meta_from`.